src/riak_kv_qry_worker.erl

```diff
@@ -187,10 +187,8 @@ make_key_conversion_fun(Table) ->
     Mod = riak_ql_ddl:make_module_name(Table),
     DDL = Mod:get_ddl(),
     fun(Key) when is_binary(Key) ->
-            {ok, DescAdjustedKey} =
-                riak_ql_ddl:negate_if_desc(
-                  tuple_to_list(sext:decode(Key)), Mod, DDL),
-            {ok, PK} = riak_ql_ddl:lk_to_pk(DescAdjustedKey, Mod, DDL),
+            {ok, PK} = riak_ql_ddl:lk_to_pk(
```

Comment:
This function is passed from a `riak_kv_index_fsm` process to a `riak_kv_vnode` to be invoked in a fold. I don't see any exception handling in a quick scan of that code path, which leads me to believe that whatever provokes that exception could bring down a vnode.